### PR TITLE
Add an extra indentation to the result of the indent function

### PR DESCRIPTION
**Pull request by mprahl**
Adding four spaces to the indentation makes the usage of `indent N`
be from the logical starting point of the resource object wrapped
in the ConfigurationPolicy.

The additional commits are to fix the unit tests so that they pass when run by CI. It's unclear why this wasn't hit previously.

This resolves:
https://github.com/open-cluster-management/backlog/issues/16040

### CODE REVIEW

File: pkg/controller/propagator/propagation_test.go [hunk 1]
```diff
@@ -26,14 +26,19 @@ func TestInitializeAttempts(t *testing.T) {
 		t.Run(
 			fmt.Sprintf(`%s="%s"`, attemptsEnvName, test.envVarValue),
 			func(t *testing.T) {
-				t.Parallel()
```
Comment (ckandag) [inline pkg/controller/propagator/propagation_test.go:29]: are these changes related to this issue, i thought you are big on keepings seperate?

File: pkg/controller/propagator/propagation.go [hunk 2]
```diff
@@ -46,7 +46,9 @@ var templateCfg templates.Config
 func Initialize(kubeconfig *rest.Config, kubeclient *kubernetes.Interface) {
 	kubeConfig = kubeconfig
 	kubeClient = kubeclient
-	templateCfg = templates.Config{StartDelim: "{{hub", StopDelim: "hub}}"}
+	// Adding four spaces to the indentation makes the usage of `indent N` be from the logical
+	// starting point of the resource object wrapped in the ConfigurationPolicy.
+	templateCfg = templates.Config{AdditionalIndentation: 4, StartDelim: "{{hub", StopDelim: "hub}}"}
```
Comment (ckandag) [inline pkg/controller/propagator/propagation.go:51]: i dont think its 4,  its 4 indent "units" ... did you test ?

File: pkg/controller/propagator/propagation_test.go (hunk 1 above)
Comment (mprahl) [reply pkg/controller/propagator/propagation_test.go:29]: @ckandag they are in a separate commit from the main chain. I needed it for the unit tests to pass in Prow.

File: pkg/controller/propagator/propagation.go (hunk 2 above)
Comment (mprahl) [reply pkg/controller/propagator/propagation.go]: @ckandag my mistake. 16 is the number that worked for me locally.

File: pkg/controller/propagator/propagation_test.go (hunk 1 above)
Comment (ckandag) [reply pkg/controller/propagator/propagation_test.go:29]: just want to note that :  these changes appear  to be  fixes/improvement to the test setup and are not really related specifically for the indent change.

**Review by ckandag — APPROVED** [review]

**Pull request merged**